src/otx/core/model/module/rotated_detection.py

```diff
@@ -38,3 +44,79 @@ def __init__(
             scheduler=scheduler,
             metric=metric,
         )
+
+    def predict_step(self, *args: torch.Any, **kwargs: torch.Any) -> InstanceSegBatchPredEntity:
+        """Predict step for rotated detection task.
+
+        Note: This method is overridden to convert masks to rotated bounding boxes.
+
+        Returns:
+            InstanceSegBatchPredEntity: The predicted polygons (rboxes), scores, labels, masks.
+        """
+        preds = super().predict_step(*args, **kwargs)
+
+        batch_scores: list[torch.Tensor] = []
+        batch_bboxes: list[tv_tensors.BoundingBoxes] = []
+        batch_labels: list[torch.LongTensor] = []
+        batch_polygons: list[list[Polygon]] = []
+        batch_masks: list[tv_tensors.Mask] = []
+
+        for img_info, pred_bboxes, pred_scores, pred_labels, pred_masks in zip(
+            preds.imgs_info,
+            preds.bboxes,
+            preds.scores,
+            preds.labels,
+            preds.masks,
+        ):
+            boxes = []
+            scores = []
+            labels = []
+            masks = []
+            polygons = []
+
+            for bbox, score, label, mask in zip(pred_bboxes, pred_scores, pred_labels, pred_masks):
+                if mask.sum() == 0:
+                    continue
+                np_mask = mask.detach().cpu().numpy().astype(int)
+                contours, hierarchies = cv2.findContours(np_mask, cv2.RETR_CCOMP, cv2.CHAIN_APPROX_SIMPLE)
+                if hierarchies is None:
+                    continue
+                rbox_polygons = []
+                for contour, hierarchy in zip(contours, hierarchies[0]):
+                    # skip inner contours
+                    if hierarchy[3] != -1 or len(contour) <= 2:
+                        continue
+                    rbox_points = Polygon(cv2.boxPoints(cv2.minAreaRect(contour)).reshape(-1))
+                    rbox_polygons.append((rbox_points, rbox_points.get_area()))
+
+                # select the largest polygon
+                if len(rbox_polygons) > 0:
+                    rbox_polygons.sort(key=lambda x: x[1], reverse=True)
+                    polygons.append(rbox_polygons[0][0])
+                    scores.append(score)
+                    boxes.append(bbox)
+                    labels.append(label)
+                    masks.append(mask)
```

Comment:
It seems polygons are updated for rotated inference, but bboxes are not. Is it okay?